### PR TITLE
[diffusion] feat: support out-of-tree torch.compile backends

### PR DESCRIPTION
**Pull request by xuzijian629**
# Problem

Diffusion compilation only selected a platform-specific backend for NPU. Other out-of-tree platforms silently used the default Inductor backend and could not provide per-module compiler options.

# Fix

Add diffusion platform hooks for selecting the `torch.compile` backend and options. Pass the DiT module to the compile helper so out-of-tree platforms can configure persistent module state. Preserve the existing Inductor mode when the platform does not provide explicit options.

# Test Plan

- Added unit coverage for custom backend and per-module option selection.
- Added coverage that explicit Inductor options replace the mutually exclusive compile mode.
- Extended the regional compilation test to verify the DiT module reaches the compile helper.
- `python3 -m compileall` on all changed Python files.
- Pinned Black 26.1.0, isort 7.0.0, Ruff 0.15.1, and codespell 2.4.1 checks passed.
- Full pytest was not run because the fresh checkout does not have the SGLang diffusion dependency set installed.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] (Not Applicable) Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #32858966820](https://github.com/sgl-project/sglang/actions/runs/32858966820)
Latest PR Test (Extra): :x: [Run #32858966302](https://github.com/sgl-project/sglang/actions/runs/32858966302)
Latest PR Test (AMD ROCm 7.2): :x: [Run #32858966723](https://github.com/sgl-project/sglang/actions/runs/32858966723)

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/platforms/interface.py [hunk 1]
```diff
@@ -123,6 +123,16 @@ class Platform:
 
     supported_quantization: list[str] = []
 
+    def get_compile_backend(self, mode: str | None = None) -> str:
+        """Return the backend used to compile diffusion modules."""
+        del mode
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/platforms/interface.py]: could we avoid the `del xxx`?

Comment (xuzijian629) [reply python/sglang/multimodal_gen/runtime/platforms/interface.py]: Reflected, thanks!

File: python/sglang/multimodal_gen/runtime/platforms/interface.py [hunk 2]
```diff
@@ -123,6 +123,15 @@ class Platform:
 
     supported_quantization: list[str] = []
 
+    def get_compile_backend(self, mode: str | None = None) -> str:
+        """Return the backend used to compile diffusion modules."""
+        return self.simple_compile_backend
+
+    def get_compile_options(self, module: torch.nn.Module) -> dict[str, object] | None:
+        """Return backend-specific options for a diffusion module."""
+        del module
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/platforms/interface.py]: ditto 😂

Comment (xuzijian629) [reply python/sglang/multimodal_gen/runtime/platforms/interface.py]: Thanks for catching this! Resubmitted a clean version

**Review by mickqian — APPROVED** [review]
